github/teams.go

```diff
@@ -32,6 +32,10 @@ type Team struct {
 	// Permission specifies the default permission for repositories owned by the team.
 	Permission *string `json:"permission,omitempty"`
 
+	// Permissions identifies the permissions that a team has on a given
+	// repository. This is only populated when calling Repositories.ListTeams.
+	Permissions *map[string]bool `json:"permissions,omitempty"`
```

Comment:
```suggestion
	Permissions map[string]bool `json:"permissions,omitempty"`
```

I see that there are two other places in this repo where we use `*map[string]bool`, but I believe that the pointer to a map is unnecessary since map is already a reference type and can be `nil` without the pointer.

Let's start here by making it just a map, and then clean up the other usage later.

Reply:
Ok, sounds good. Thanks @gmlewis!